.github/workflows/deployment.yaml

```diff
@@ -0,0 +1,52 @@
+name: deployment
+
+on:
+  pull_request:
+    branches: [main]
+  push:
+    branches: [main]
+
+jobs:
+  checks:
+    if: github.event_name != 'push'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '12.x'
+      - name: Test Build
+        run: |
+          if [ -e yarn.lock ]; then
+            yarn install --frozen-lockfile
+          elif [ -e package-lock.json ]; then
+            npm ci
+          else
+            npm i
+          fi
+          yarn build
+  gh-release:
+    if: github.event_name != 'pull_request'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '12.x'
+      - uses: webfactory/ssh-agent@v0.5.0
+        with:
+          ssh-private-key: ${{ secrets.GH_PAGES_DEPLOY }}
```

Comment:
I think we'll need a repo owner to add a public key to Github deploy keys and a private key to Github secrets so Docusaurus can run the deployment for us.
CC @humphd

Reply:
I agree, but it seems you need to add it to docker setup if cause I'm trying it on act-cli running it on local

Reply:
also local .env file like in Telescope like a for the search feature as well for my PR

Reply:
Once we're ready to merge this, we can do that

Reply:
> I think we'll need a repo owner to add a public key to Github deploy keys and a private key to Github secrets so Docusaurus can run the deployment for us. CC @humphd



> Nice! What's the URL for the live version it pushed to @trangntt-016's repo? When this completes the end product should be a built version that's live GH Pages. I think there's still more steps to do.

Hi @humphd ,

thanks for your feedback. Agree with @DukeManh , and I also tested on my `test` branch and it worked after I added an SSH key in my repo's setting. I wonder if you could please generate a secret key `GH_PAGES_DEPLOY` and add it to `ICTOER` setting?

Reply:
Can I ask that you document the steps that need to be done in `CONTRIBUTING.md`, and we can do that.  I want to make sure it's clear for future maintainers what needs to happen vs. just doing it once.